### PR TITLE
Show Application Use Case Task

### LINKED ISSUE #39: Show Application Task
Use case 5.5

- Backend api routes for getting applicant merits and availability
- Frontend api call for getting applicant merits and availability

### PR DESCRIPTION
**Pull request by willeasp**
Fixes #39 

**Frontend**
- Added modal (popup) when a user is clicked. 
- Applicant data retrieved from server
- Buttons accept and reject not doing anything
- Competence names and id mapping currently stored in frontend but could easily be made an api call

**Backend**
- Added path for applicant data which retrieves both *availability* and *competence*

### CODE REVIEW

File: frontend/src/components/ApplicantTable.vue [hunk 1]
```diff
@@ -50,20 +117,57 @@ export default {
...
+                this.viewApplicant = this.list[0];
             } catch (error) {
                 this.loadingError = true;
             }
+        },
+        async onRowSelected ([selected]) {
+            /* show the popup window */
+            if (selected) {
+                try {
+                    const data = await applicants.getApplication(selected.id);
```
Comment (virsp) [inline frontend/src/components/ApplicantTable.vue:134]: Should we make this go through the store, so we keep consistency?

File: frontend/src/components/ApplicantTable.vue [hunk 2]
```diff
@@ -1,36 +1,92 @@
...
+                            <b-button size="lg" variant="danger" @click="rejectApplicant()">
+                                Reject
+                            </b-button>
+                        </template>
+                    </b-modal>
+
+                </b-row>
+            </b-container>
+        </b-row>
+        <b-row>
```
Comment (virsp) [inline frontend/src/components/ApplicantTable.vue:76]: Maybe make this a separate component, we can discuss on meeting

File: recruitmentApplication/src/main/java/se/kth/iv1201/group6/recruitmentApplication/repository/AvailabilityRepository.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import se.kth.iv1201.group6.recruitmentApplication.model.Availability;
+
+/**
+ * Simple CRUD repository for availability entity
+ */
+@Transactional
```
Comment (simonljohannesson) [inline recruitmentApplication/src/main/java/se/kth/iv1201/group6/recruitmentApplication/repository/AvailabilityRepository.java:16]: Transactional( MANDATORY ) something here

File: recruitmentApplication/src/main/java/se/kth/iv1201/group6/recruitmentApplication/repository/CompetenceProfileRepository.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import se.kth.iv1201.group6.recruitmentApplication.model.CompetenceProfile;
+
+/**
+ * Simple CRUD repository for CompetenceProfile entity
+ */
+@Repository
+@Transactional
```
Comment (simonljohannesson) [inline recruitmentApplication/src/main/java/se/kth/iv1201/group6/recruitmentApplication/repository/CompetenceProfileRepository.java:16]: MANDATORY

File: recruitmentApplication/src/main/java/se/kth/iv1201/group6/recruitmentApplication/service/ApplicantService.java [hunk 5]
```diff
@@ -62,4 +78,20 @@ public void createApplicant(CreateApplicantDto applicant) throws ApplicantConfli
...
     }
+
+    /**
+     * Get an applicants data concerning an application
+     * @param applicantId The id of the applicant
+     * @return ApplicantDataDto competences and availability
+     */
+    @Transactional
+    public ApplicantDataDto getApplicantData(Long applicantId) {
+        var availability = availabilityRepository.findAvailability(applicantId);
```
Comment (simonljohannesson) [inline recruitmentApplication/src/main/java/se/kth/iv1201/group6/recruitmentApplication/service/ApplicantService.java:89]: On illegal id, receive emtpy array or internal server error

**Comment by virsp** [comment]
Suggested changes made and functioning as intended merging

**Pull request merged**